### PR TITLE
789 - Fix selector to show correct network for current env

### PR DESCRIPTION
**Pull request by giladt**
# BUG fixed:
The network selector dropdown should now show the correct network even if previously a local-storage item was set.

### CODE REVIEW

File: src/resources/elements/NetworkFeedback/NetworkFeedback.ts [hunk 1]
```diff
@@ -26,9 +26,10 @@ export class NetworkFeedback {
     const isTestnet = [Networks.Rinkeby, Networks.Alfajores, Networks.Kovan].includes(storedNetwork);
     if (
       storedNetwork &&
-      (this.isProduction && !isMainnet) || (!this.isProduction && !isTestnet)
+      (this.isProduction && isTestnet) || (!this.isProduction && isMainnet)
```
Comment (hiaux0) [inline src/resources/elements/NetworkFeedback/NetworkFeedback.ts:29]: 1.How can `this.isProduction && isTestnet` ever be true?

2. Can we have more descriptives if-statements in the future?
```ts
const A = this.isProduction && isTestnet
const B = !this.isProduction && isMainnet
const C = A || B
const D = storedNetwork && C
```

File: src/resources/elements/NetworkFeedback/NetworkFeedback.ts [hunk 2]
```diff
@@ -26,9 +26,10 @@ export class NetworkFeedback {
...
     ) {
-      this.storageService.lsRemove("network");
+      this.network = this.isProduction ? Networks.Mainnet : Networks.Rinkeby;
```
Comment (hiaux0) [inline src/resources/elements/NetworkFeedback/NetworkFeedback.ts:31]: what about Celo?

**Review by hiaux0 — CHANGES_REQUESTED** [review]

File: src/resources/elements/NetworkFeedback/NetworkFeedback.ts (hunk 2 above)
Comment (giladt) [reply src/resources/elements/NetworkFeedback/NetworkFeedback.ts]: This one is for the case there's no selection, or selection is illegal (e.g. rinkeby on mainnet).

File: src/resources/elements/NetworkFeedback/NetworkFeedback.ts [hunk 3]
```diff
@@ -20,15 +20,20 @@ export class NetworkFeedback {
...
+      locallyStoredNetwork &&
+      ((this.isProductionEnv && isLocallyTestnet) || (!this.isProductionEnv && isLocallyMainnet))
     ) {
-      this.storageService.lsRemove("network");
+      /**
+      * If stored network is illegal, should be treated as if not selected yet,
+      * thus Ethereum network is the default.
+      */
+      this.network = this.isProductionEnv ? Networks.Mainnet : Networks.Rinkeby;
+      this.storageService.lsSet("network", `${this.network}`);
```
Comment (hiaux0) [inline src/resources/elements/NetworkFeedback/NetworkFeedback.ts:36]: Don't need comment, descriptive var name enough.

```suggestion
      const defaultNetwork = this.isProductionEnv ? Networks.Mainnet : Networks.Rinkeby;
      this.network = defaultNetwork
      this.storageService.lsSet("network", `${this.network}`);
```

File: src/resources/elements/NetworkFeedback/NetworkFeedback.ts [hunk 4]
```diff
@@ -20,15 +20,20 @@ export class NetworkFeedback {
...
+    this.isProductionEnv = process.env.NODE_ENV !== "development" || process.env.NETWORK === Networks.Mainnet;
+    const locallyStoredNetwork = this.storageService.lsGet<AllowedNetworks>("network");
+    const isLocallyMainnet = [Networks.Mainnet, Networks.Celo, Networks.Arbitrum].includes(locallyStoredNetwork);
+    const isLocallyTestnet = [Networks.Rinkeby, Networks.Alfajores, Networks.Kovan].includes(locallyStoredNetwork);
     if (
-      storedNetwork &&
-      (this.isProduction && !isMainnet) || (!this.isProduction && !isTestnet)
+      locallyStoredNetwork &&
+      ((this.isProductionEnv && isLocallyTestnet) || (!this.isProductionEnv && isLocallyMainnet))
     ) {
```
Comment (hiaux0) [inline src/resources/elements/NetworkFeedback/NetworkFeedback.ts:30]:
```ts
const invalidlyStoredTestnet = this.isProductionEnv && isLocallyTestnet
const invalidlyStoredMainnet = !this.isProductionEnv && isLocallyMainnet
const illegalNetwork = locallyStoredNetwork && (invalidlyStoredTestnet || invalidlyStoredMainnet);
if (illegalNetwork) {}
```

**Review by hiaux0 — CHANGES_REQUESTED** [review]

File: src/resources/elements/NetworkFeedback/NetworkFeedback.ts [hunk 5]
```diff
@@ -20,15 +20,21 @@ export class NetworkFeedback {
...
-    const isTestnet = [Networks.Rinkeby, Networks.Alfajores, Networks.Kovan].includes(storedNetwork);
-    if (
-      storedNetwork &&
-      (this.isProduction && !isMainnet) || (!this.isProduction && !isTestnet)
-    ) {
-      this.storageService.lsRemove("network");
+    this.isProductionEnv = process.env.NODE_ENV !== "development" || process.env.NETWORK === Networks.Mainnet;
+    const locallyStoredNetwork = this.storageService.lsGet<AllowedNetworks>("network");
+    if (locallyStoredNetwork) {
+      const defaultNetwork = this.isProductionEnv ? Networks.Mainnet : Networks.Rinkeby;
```
Comment (hiaux0) [inline src/resources/elements/NetworkFeedback/NetworkFeedback.ts:26]: future: put variable closer to where it is used.
why was it put before the `if (illegalNetwork) {` block?

**Review by hiaux0 — APPROVED** [review]

**Pull request merged**